linchpin/provision/library/dummy.py

```diff
@@ -79,7 +82,7 @@ def __init__(self):
 #        return host_str
 
 
-    def allocate(self, name, count=1):
+    def allocate(self, name, count=1, domain=""):
```

Comment:
Since domain option is given as a parameter to dummy module . User can always default it with jinja in Pinfile. Instead of we adding it at module level.